src/main/kotlin/io/hsar/wh40k/combatsimulator/model/World.kt

```diff
@@ -29,13 +22,74 @@ data class World(
                 .distanceToPosition(unitPositions.getValue(otherUnit))
     }
 
-    fun canMoveToUnit(unit: UnitInstance, otherUnit: UnitInstance, moveType: MoveAction): Boolean {
-        return (distanceApart(unit, otherUnit) - 1 <=
-                unit.unit.stats.baseStats.getValue(BaseStat.AGILITY).getBonus()
-                        .let { bonus ->
-                            moveType.getMovementRange(bonus)
-                        })
-                && moveType.isValidMovementPath(unitPositions.getValue(unit), unitPositions.getValue(otherUnit))
+    /*
+    Given a user and another UnitInstance that the user wants to move towards, this will handle the low level
+    mechanics of moving the unit as close as it can get for its movement range, stoppign if it comes within melee
+    distance
+     */
+    fun moveTowards(unit: UnitInstance, otherUnit: UnitInstance, maxMovement: Int): Unit {
```

Comment:
This implementation is... fairly fugly. I think I'll leave it be for now but this is a candidate for some serious reworking in the near future.

Reply:
I agree it's not functional and probably not the most efficient either. Should be an easy refactor in place